src/futures/write/generic/decoder.rs

```diff
@@ -182,3 +182,21 @@ impl<W: AsyncWrite, D: Decode> AsyncWrite for Decoder<W, D> {
         }
     }
 }
+
+impl<W: AsyncWrite + AsyncRead, D: Decode> AsyncRead for Decoder<W, D> {
```

Comment:
```suggestion
impl<W: AsyncRead, D: Decode> AsyncRead for Decoder<W, D> {
```

Reply:
> without `AsyncWrite` there is no `get_pin_mut()`, even if I do `self.project().writer.get_pin_mut().poll_read(cx, buf)` as `BufWriter` implementation also requires `AsyncWrite`.

Thanks, can we update this function so that it doesn't require that bound?

The function in question does not use `AsyncWrite` in anyway.

https://github.com/Nullus157/async-compression/blob/9d9fe02be78a8a074816e1ad08b2c72ffe3ef137/src/futures/write/buf_writer.rs#L132

Reply:
Thanks!